tests/by-util/test_cp.rs

```diff
@@ -2660,6 +2660,71 @@ fn test_copy_through_dangling_symlink_no_dereference() {
         .no_stdout();
 }
 
+#[test]
+fn test_cp_symlink_overwrite_detection() {
+    let ts = TestScenario::new(util_name!());
+    let at = &ts.fixtures;
+
+    at.mkdir("good");
+    at.mkdir("tmp");
+    at.write("README", "file1");
+    at.write("good/README", "file2");
+
+    ts.ccmd("ln")
+        .arg("-s")
+        .arg("foo")
+        .arg("tmp/README")
+        .succeeds();
+
+    at.touch("tmp/foo");
+
+    ts.ucmd()
+        .arg("README")
+        .arg("good/README")
+        .arg("tmp")
+        .fails()
+        .stderr_only(if cfg!(target_os = "windows") {
+            "cp: will not copy 'good/README' through just-created symlink 'tmp\\README'\n"
+        } else if cfg!(target_os = "macos") {
+            "cp: will not overwrite just-created 'tmp/README' with 'good/README'\n"
+        } else {
+            "cp: will not copy 'good/README' through just-created symlink 'tmp/README'\n"
+        });
+    let contents = at.read("tmp/foo");
+    // None of the files seem to be copied in macos
+    if cfg!(not(target_os = "macos")) {
+        assert_eq!(contents, "file1");
+    }
+}
+
+#[test]
+fn test_cp_dangling_symlink_inside_directory() {
+    let ts = TestScenario::new(util_name!());
+    let at = &ts.fixtures;
+
+    at.mkdir("good");
+    at.mkdir("tmp");
+    at.write("README", "file1");
+    at.write("good/README", "file2");
+
+    ts.ccmd("ln")
+        .arg("-s")
+        .arg("foo")
+        .arg("tmp/README")
+        .succeeds();
```

Comment:
```suggestion
    at.symlink_file("foo", "tmp/README");
```
Same in the other places.